taca/delivery/delivery_classes.py

```diff
@@ -111,3 +137,251 @@ def stage_data(self):
+            add_user, ignore_orderportal_members
+        )
+        self.project_description = self.get_project_description(project_description)
+
+    def get_order_details(self):
+        """Fetch order details from order portal"""
+        projects_db = self.status_db_connection.connection["projects"]
+        view = projects_db.view("order_portal/ProjectID_to_PortalID")
+        rows = view[self.project_id].rows
+        if len(rows) < 1:
+            raise AssertionError(f"Project {self.project_id} not found in StatusDB")
+        if len(rows) > 1:
+            raise AssertionError(
+                f"Project {self.project_id} has more than one entry in StatusDB orderportal_db"
+            )
+        portal_id = rows[0].value
+        # Get project info from order portal API
+        get_project_url = "{}/v1/order/{}".format(
+            self.orderportal.get("orderportal_api_url"), portal_id
+        )
+        headers = {
+            "X-OrderPortal-API-key": self.orderportal.get("orderportal_api_token")
+        }
+        response = requests.get(get_project_url, headers=headers)
+        if response.status_code != 200:
+            raise AssertionError(
+                "Status code returned when trying to get "
+                "project info from the order portal: "
+                f"{portal_id} was not 200. Response was: {response.content}"
+            )
+        return json.loads(response.content)
+
+    def get_pi_email(self, given_pi_email):
+        """Determine the PI email."""
+        if given_pi_email:
+            logger.warning(
+                f"PI email for project {self.project_id} specified by user: {given_pi_email}"
+            )
+            return given_pi_email
+        else:
+            found_pi_email = self.order_details["fields"]["project_pi_email"]
+            logger.info(
+                f"PI email for project {self.project_id} found: {found_pi_email}"
+            )
+            return found_pi_email
+
+    def get_other_member_details(
+        self, other_member_emails=[], ignore_orderportal_members=False
+    ):
+        """Set other contact details if available. This is not mandatory so
+        the method will not raise error if it could not find any contact
+        """
+        other_member_details = []
+        if not ignore_orderportal_members:
+            logger.info("Fetching additional members from order portal.")
+            if self.order_details.get("owner"):
+                owner_email = self.order_details.get("owner").get("email")
+                if (
+                    owner_email
+                    and owner_email != self.pi_email
+                    and owner_email not in other_member_details
+                ):
+                    other_member_details.append(owner_email)
+            if self.order_details.get("fields"):
+                bioinfo_email = self.order_details.get("fields").get("project_bx_email")
```

Comment:
Maybe add a default, so it won't throw an exception if there is ever a case that `fields` doesn't exist?
```suggestion
                bioinfo_email = self.order_details.get("fields", {}).get("project_bx_email")
```

Reply:
If `fields` doesn't exist, it won't enter the `if`. But yeah, I should add it to some other places I think.